sdk/kubeflow/trainer/api/trainer_client_abc.py

```
@@ -0,0 +1,62 @@
+# Copyright 2024 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Optional, Dict, List
+
+from kubeflow.trainer.constants import constants
+from kubeflow.trainer.types import types
+
+
+class TrainerClientABC(ABC):
+    @abstractmethod
+    def delete_job(self, name: str):
+        pass
+
+    @abstractmethod
+    def get_job(self, name: str) -> types.TrainJob:
+        pass
```

should we raise an error here ?

raise `NotImplementedError` ?
and similarly in other functions

---

Abstract classes cannot be instantiated, so these methods can never be called (so the `NotImplementedError`) would never be raised. I did a quick search and any other ABC examples I found used the `pass` keyword.

---

>A class that has a metaclass derived from ABCMeta cannot be instantiated unless all of its abstract methods and properties are overridden. 

https://docs.python.org/3/library/abc.html#abc.abstractmethod